lms/djangoapps/grades/signals/handlers.py

```diff
@@ -199,6 +202,28 @@ def recalculate_course_grade(sender, course, course_structure, user, **kwargs):
     Updates a saved course grade.
     """
     CourseGradeFactory().update(user, course, course_structure)
+    grade = CourseGradeFactory()._get_saved_grade(user, course, course_structure)
+    if grade._percent == 1.0:
+        course_enrollment = CourseEnrollment.objects.get(course_id=course.id, user=user.id)
+        persist_course_grade = PersistentCourseGrade.objects.get(user_id=user.id, course_id=course.id)
+        duration = persist_course_grade.passed_timestamp - course_enrollment.created if persist_course_grade.passed_timestamp else None
+        import pdb; pdb.set_trace()
+        data = {
+            "contentProvider": "FastLane",
+            "userId": user.id,
+            "courseId": course.id.to_deprecated_string(),
+            "lastlogin": user.last_login,
+            "percentageOfcompletion": grade.percent,
+            "duration": duration,
+            "lastVisit": '---',
+            "completationDate" : persist_course_grade.passed_timestamp,
```

Comment:
Can it be retrieved from `grade` not from `persist_course_grade ` ?

Reply:
`grade` does not contain an attribute `passed_timestamp`